### PR TITLE
Add support for WITHIN GROUP for ARRAY_AGG and LISTAGG functions

### PR DESCRIPTION
**Pull request by bishwajit-db**
Add support for WITHIN GROUP for ARRAY_AGG and LISTAGG functions. Fixes issue #121

### CODE REVIEW

**Comment by sundarshankar89** [comment]
@nsenno-dbr FYA.

File: src/databricks/labs/remorph/snow/databricks.py [hunk 1]
```diff
@@ -299,6 +293,38 @@ def _parse_date_trunc(self: Databricks.Generator, expression: local_expression.D
...
+    within_group: exp.WithinGroup,
+) -> WithinGroupParams:
+    has_distinct = isinstance(expr.this, exp.Distinct)
+    agg_col = expr.this.expressions[0] if has_distinct else expr.this
+    order_expr = within_group.expression
+    order_col = order_expr.expressions[0].this
+    desc = order_expr.expressions[0].args.get("desc")
+    is_order_asc = not desc or exp.false() == desc
+    if has_distinct and agg_col != order_col:
+        raise ParseError("If both DISTINCT and WITHIN GROUP are specified, both must refer to the same column.")
```
Comment (nsenno-dbr) [inline src/databricks/labs/remorph/snow/databricks.py:307]: Can we leave a TODO here so that we can relax this condition in the future

Comment (bishwajit-db) [reply src/databricks/labs/remorph/snow/databricks.py:307]: @nsenno-dbr This restriction is from Snowflake as mentioned in https://docs.snowflake.com/en/sql-reference/functions/array_agg#usage-notes

Comment (bishwajit-db) [reply src/databricks/labs/remorph/snow/databricks.py:302]: Added a TODO comment

File: src/databricks/labs/remorph/snow/databricks.py [hunk 2]
```diff
@@ -34,6 +35,13 @@
 }
 
 
+@dataclass
```
[the remark this answers is not in the document]
Comment (bishwajit-db) [reply src/databricks/labs/remorph/snow/databricks.py]: Moved

**Comment by nsenno-dbr** [comment]
LGTM

**Review by nsenno-dbr — APPROVED** [review]

**Pull request merged**